### PR TITLE
HEAL 69, 70, 71, 79, 80 

### PR DESCRIPTION
**Pull request by hasinireddy23**
HEAL 69 - Frontend Implementation to show all the appointments in the Manage Medical Records tab with an option to filter past, upcoming and all appointments
HEAL 70 - Implemented pagination for the Manage Medical Records tab where all the confirmed appointments are shown.
HEAL 71 - Created and Deployed Smart Contracts for Creating, approving and declining medical records request from doctors to patients
HEAL 79 - Front end Implementation to request and upload medical records to patient by doctor in Manage Medical Records tab
HEAL 80 - Patient Should be able to receive a Email Notification whenever a medical records access request is raised by doctor from Manage Medical Records tab

image

image

image

image

### CODE REVIEW

File: client/src/components/TopNav.tsx [hunk 1]
```diff
@@ -1,59 +1,107 @@
-
-import React, { FunctionComponent, useContext } from "react";
-import { Link } from 'react-router-dom'
-import { AuthConsumer, AuthContext } from "../providers/AuthProvider";
-import WalletComponent from "./WalletComponent";
+import React, { FunctionComponent, useContext } from 'react';
```
Comment (varunalla) [inline client/src/components/TopNav.tsx]: install prettier and fix formatting

File: client/src/providers/HealthProvider.tsx [hunk 2]
```diff
@@ -133,6 +136,60 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
     }
   };
 
+  const requestMedicalRecordHealthBlockContract = async (patientAddress: string) => {
```
Comment (varunalla) [inline client/src/providers/HealthProvider.tsx]: remove console.log statements and try returning the info in the method to validate

File: package.json [hunk 3]
```diff
@@ -31,5 +32,8 @@
     "generate-types": "typechain --target=ethers-v5 'client/src/contracts/*.json' --out-dir client/src/contracts",
     "migrate": "tsc -p ./tsconfig.migrate.json --outDir ./migrations && truffle migrate",
     "typecheck": "truffle compile && pnpm generate-types && tsc --noEmit"
+  },
+  "devDependencies": {
+    "@types/react-paginate": "^7.1.1"
```
Comment (varunalla) [inline package.json]: can you explain why this is dev dependencies

**Review by varunalla — CHANGES_REQUESTED** [review]
Thank you for the review. The changes in ManageMedicalRecords in pascal casing,

File: package.json (hunk 3 above)
Comment (hasinireddy23) [reply package.json:37]: When I tried to install react-paginate, there was some error. So, I tried to resolve that error. Then this got created.

File: client/src/providers/HealthProvider.tsx (hunk 2 above)
Comment (hasinireddy23) [reply client/src/providers/HealthProvider.tsx]: > remove console.log statements and try returning the info in the method to validate

Removed console.log statements

File: package.json (hunk 3 above)
Comment (hasinireddy23) [reply package.json]: > can you explain why this is dev dependencies

When I tried to install react-paginate, there was some error. So, I tried to resolve that error. Then this got created.

File: client/src/components/TopNav.tsx (hunk 1 above)
Comment (hasinireddy23) [reply client/src/components/TopNav.tsx]: > install prettier and fix formatting

Fixed it

File: client/src/providers/HealthProvider.tsx (hunk 2 above)
Comment (hasinireddy23) [reply client/src/providers/HealthProvider.tsx]: removed console.log statements

**Review by varunalla — APPROVED** [review]

**Review by Shruthi23 — CHANGES_REQUESTED** [review]
Please remove all console.log statements.

File: client/src/components/Doctor/ManageMedicalrecords.tsx [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+        patientName,
+        patientEmail,
+      );
+      alert('Medical record request sent successfully');
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  // function sendRequestEmail(to: string) {
```
Comment (varunalla) [inline client/src/components/Doctor/ManageMedicalrecords.tsx]: remove commented code if it is not necessary

File: client/src/components/Patient/PatientManageMedicalRecords.tsx [hunk 5]
```diff
@@ -0,0 +1,427 @@
+import React, { FunctionComponent, useContext, useState ,useEffect} from 'react';
+import ReactPaginate from 'react-paginate';
+import { HealthContext } from '../../providers/HealthProvider';
+import { AuthContext } from '../../providers/AuthProvider';
+import AWS from 'aws-sdk';
```
Comment (varunalla) [inline client/src/components/Patient/PatientManageMedicalRecords.tsx]: why is this needed on the front end?

File: client/src/components/Patient/PatientManageMedicalRecords.tsx [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+    }
+  };
+
+  useEffect(()=>{
+    if(currentAccount){
+    GetAllMedicalRecordRequests(currentAccount);
+  }
+  },[])  
+
+  // function sendAcceptEmail(to: string) {
```
Comment (varunalla) [inline client/src/components/Patient/PatientManageMedicalRecords.tsx]: remove commented code if not used

File: client/src/components/Patient/PatientManageMedicalRecords.tsx [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+      GetAllMedicalRecordRequests(currentAccount);}
+      alert('Medical record request rejected');
+    }catch(err){
+      console.log("Error while rejecting the medical Record Requests",err);
+    }
+  };
+
+  useEffect(()=>{
+    if(currentAccount){
+    GetAllMedicalRecordRequests(currentAccount);
```
Comment (varunalla) [inline client/src/components/Patient/PatientManageMedicalRecords.tsx]: why does this exists? where is the result stored?

File: client/src/components/Patient/PatientManageMedicalRecords.tsx [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+        GetAllMedicalRecordRequests(currentAccount);}
+    }catch(err){
+      console.log("Error while approving the medical Record Requests",err);
+    }
+  };
+  const rejectRequest=async(requestID: string)=>{
+    try{
+      await rejectMedicalRecordsRequestHealthBlockContract?.(requestID)
+      if(currentAccount){
+      GetAllMedicalRecordRequests(currentAccount);}
```
Comment (varunalla) [inline client/src/components/Patient/PatientManageMedicalRecords.tsx]: why does this exists? where is the result stored?

File: client/src/components/Patient/PatientManageMedicalRecords.tsx [hunk 9]
```diff
@@ -0,0 +1,427 @@
...
+    } catch (err) {
+      console.log("Error while fetching the medical Record Requests",err);
+    }
+  };
+
+  const approveRequest=async(requestID: string)=>{
+    try{
+      await approveMedicalRecordsRequestHealthBlockContract?.(requestID)
+      if(currentAccount){
+        GetAllMedicalRecordRequests(currentAccount);}
```
Comment (varunalla) [inline client/src/components/Patient/PatientManageMedicalRecords.tsx]: why does this exists? where is the result stored?

File: client/src/providers/HealthProvider.tsx [hunk 10]
```diff
@@ -194,6 +220,91 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
...
+    patientAddress: string,
+    patientName: string,
+    docEmail: string,
+    patientEmail: string,
+  ) => {
+      try {
+        console.log("patientAddress",patientAddress);
+        console.log("patientName",patientName);
+        console.log("docEmail",docEmail);
+        console.log("patientEmail",patientEmail);
```
Comment (varunalla) [inline client/src/providers/HealthProvider.tsx]: are these needed?

**Review by varunalla — CHANGES_REQUESTED** [review]

File: client/src/providers/HealthProvider.tsx (hunk 10 above)
Comment (hasinireddy23) [reply client/src/providers/HealthProvider.tsx]: Deleted them

File: client/src/components/Patient/PatientManageMedicalRecords.tsx (hunk 6 above)
Comment (hasinireddy23) [reply client/src/components/Patient/PatientManageMedicalRecords.tsx]: That code was working but that is in frontend so I commented for now. Shruthi is fixing that ses service to go backend. Will remove this after I get shruthi's changes.

File: client/src/components/Patient/PatientManageMedicalRecords.tsx (hunk 5 above)
Comment (hasinireddy23) [reply client/src/components/Patient/PatientManageMedicalRecords.tsx]: removed aws-sdk import. Also I'm not using react-paginate. So removed that as well.

File: client/src/components/Doctor/ManageMedicalrecords.tsx (hunk 4 above)
Comment (hasinireddy23) [reply client/src/components/Doctor/ManageMedicalrecords.tsx]: That code was working but that is in frontend so I commented for now. Shruthi is fixing that ses service to go backend. Will remove this after I get shruthi's changes.

File: client/src/components/Patient/PatientManageMedicalRecords.tsx (hunk 7 above)
Comment (hasinireddy23) [reply client/src/components/Patient/PatientManageMedicalRecords.tsx]: To get all the medical record requests list. The response is being set to medicalRecordsRequests state to display the medical records.

File: client/src/components/Patient/PatientManageMedicalRecords.tsx (hunk 8 above)
Comment (hasinireddy23) [reply client/src/components/Patient/PatientManageMedicalRecords.tsx]: To get all the medical record requests list. The response is being set to medicalRecordsRequests state to display the medical records.

File: client/src/components/Patient/PatientManageMedicalRecords.tsx (hunk 9 above)
Comment (hasinireddy23) [reply client/src/components/Patient/PatientManageMedicalRecords.tsx]: To get all the medical record requests list. The response is being set to medicalRecordsRequests state to display the medical records.

**Review by varunalla — APPROVED** [review]

**Review by Shruthi23 — APPROVED** [review]

**Review by Shruthi23 — APPROVED** [review]

**Review by Shruthi23 — APPROVED** [review]

**Review by Shruthi23 — APPROVED** [review]

**Review by Shruthi23 — APPROVED** [review]
